src/docker/install.ts

```diff
@@ -92,6 +93,26 @@ export class Install {
       extractFolder = path.join(extractFolder, 'docker');
     }
     core.debug(`docker.Install.download extractFolder: ${extractFolder}`);
+    return extractFolder;
+  }
+
+  public async download(): Promise<string> {
+    let extractFolder: string;
+
+    core.info(`Downloading Docker ${this.version} from ${this.channel}`);
+
+    this._version = this.version;
```

Comment:
Oh, one thing I missed:

https://github.com/docker/actions-toolkit/blob/2941b118856faf3cc0111886b7d64ebabf3e34ff/src/docker/install.ts#L128

This seems to be used as a "version" in the cache, so `master` would never be invalidated?
Using the manifest digest would probably be our best bet here, but the documentation for `cacheDir` mentions that the version should be in a semver format..

Reply:
Oh, I think we don't actually make use of this cache for docker binaries, so I think we can just remove this step.

The only place where we use the tc cache is for image pull: 

https://github.com/docker/actions-toolkit/blob/2eba5d9d103d5a314b8dcfada334a3cca52459ad/src/docker/docker.ts#L160-L184
https://github.com/docker/actions-toolkit/blob/2eba5d9d103d5a314b8dcfada334a3cca52459ad/src/cache.ts#L79-L80

Reply:
Yes atm this is only for the `pull` method. the hosted tool cache semantic is a bit tricky but we could rely on go mod syntax for semver like `v0.0.0-20230405223818-a090f58aa992`.